File: coauthor-for-vs-code/src/latex/texTools.ts
// Standard library imports
import * as path from 'path';

// Local imports - log
import * as logger from '../logger/logUtils';

// Local imports - utilities
import { executeCommand } from '../utils/execUtils';

const CHANNEL = 'LaTeXCommands';
logger.initialize(CHANNEL);

/**
 * Compile a LaTeX file to PDF
 * @param latexFile Path to the LaTeX file
 * @param channel Optional channel for logging
 * @returns Promise<boolean> True if compilation succeeded
 */
export async function compileLatex2Pdf(
  latexFile: string,
  channel: string = CHANNEL,
): Promise<boolean> {
  try {
    const outputDirectory = path.dirname(latexFile);
    const command = [
      'pdflatex',
      '-interaction=nonstopmode',
      `-output-directory="${outputDirectory}"`,
      `"${latexFile}"`,
    ];

    const result = await executeCommand(command, { channel });
    if (result.success) {
      logger.info(channel, `Successfully compiled ${latexFile}`);
      return true;
    }
    return false;
  } catch (err) {
    logger.error(
      channel,
      `Error compiling LaTeX: ${err instanceof Error ? err.message : String(err)}`,
    );
    return false;
  }
}

File: coauthor-python/coauthor/latex/tex_tools.py
import os

from ..logger import logger
from ..utils.exec import executeCommand


def compile_latex_to_pdf(latexFile: str) -> bool:
    """Compile a LaTeX file to PDF using pdflatex."""
    outputDirectory = os.path.dirname(latexFile)
    command = ["pdflatex", "-interaction=nonstopmode", "-output-directory=" + outputDirectory, latexFile]

    try:
        success, stdout, stderr = executeCommand(command, captureOutput=True)
        if success:
            logger.info(f"Compiled {latexFile} successfully.")
            return True
        else:
            logger.error(f"Error compiling {latexFile}")
            return False
    except ValueError as e:
        logger.error(f"Error compiling {latexFile}: {str(e)}")
        return False
